src/V3Ast.cpp

```diff
@@ -93,7 +93,10 @@ AstNode* AstNode::abovep() const {
     // m_headtailp only valid at beginning or end of list
     // Avoid supporting at other locations as would require walking
     // list which is likely to cause performance issues.
-    UASSERT_OBJ(!m_nextp || firstAbovep(), this, "abovep() not allowed when in midlist");
```

Comment:
We should probably avoid modifying core AST functions unless strictly necessary. Could we not do this differently?

Reply:
I agree, I restored old implementation